extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -58,13 +55,51 @@ public YoutubeMixPlaylistExtractor(final StreamingService service,
     @Override
     public void onFetchPage(@Nonnull final Downloader downloader)
             throws IOException, ExtractionException {
-        final String url = getUrl() + "&pbj=1";
-        final Response response = getResponse(url, getExtractorLocalization());
-        final JsonArray ajaxJson = JsonUtils.toJsonArray(response.responseBody());
-        initialData = ajaxJson.getObject(3).getObject("response");
+        final Localization localization = getExtractorLocalization();
+        final URL url = stringToURL(getUrl());
+        final String mixPlaylistId = getId();
+        final String videoId = getQueryValue(url, "v");
+        final String playlistIndexString = getQueryValue(url, "index");
+
+        final byte[] body;
+        if (videoId != null) {
+            if (playlistIndexString != null) {
+                body = JsonWriter.string(prepareJsonBuilder(localization,
+                        getExtractorContentCountry())
+                        .value("videoId", videoId)
+                        .value("playlistId", mixPlaylistId)
+                        .value("playlistIndex", Integer.parseInt(playlistIndexString))
+                        .done())
+                        .getBytes(UTF_8);
+            } else {
+                body = JsonWriter.string(prepareJsonBuilder(localization,
+                        getExtractorContentCountry())
+                        .value("videoId", videoId)
+                        .value("playlistId", mixPlaylistId)
+                        .done())
+                        .getBytes(UTF_8);
+            }
+        } else {
+            body = JsonWriter.string(prepareJsonBuilder(localization,
+                    getExtractorContentCountry())
+                    .value("playlistId", mixPlaylistId)
+                    .done())
+                    .getBytes(UTF_8);
+        }
```

Comment:
Untested but should work: refactoring

```suggestion
        final JsonBuilder<JsonObject> jsonBody = prepareJsonBuilder(localization,
                getExtractorContentCountry())
                .value("playlistId", mixPlaylistId)
                .end();

        if (videoId != null) {
            jsonBody.value("videoId", videoId);
        }

        if (playlistIndexString != null) {
            jsonBody.value("playlistIndex", Integer.parseInt(playlistIndexString));
        }

        final byte[] body = JsonWriter.string(jsonBody).getBytes(UTF_8);
```

Reply:
Thank you! I tried to search a way to make this shorter but I didn't know how to do it. I will apply your suggestion.